src/Driver/SolverTypes/MISSolverType.jl

```diff
@@ -148,21 +170,66 @@ function solversetup(
     # slower processes (advection and diffusion)
     if ode_solver.discrete_splitting
         numerical_flux_first_order =
-            (dg.numerical_flux_first_order, (dg.numerical_flux_first_order,))
+            (ode_solver.numerical_flux_first_order_slow, (ode_solver.numerical_flux_first_order_slow,))
     else
-        numerical_flux_first_order = dg.numerical_flux_first_order
+        numerical_flux_first_order = ode_solver.numerical_flux_first_order_slow
     end
     slow_dg = remainder_DGModel(
         dg,
         (fast_dg,);
         numerical_flux_first_order = numerical_flux_first_order,
     )
 
+    if ode_solver.hivi_splitting
+        fast_dg_h = DGModel(
+            fast_model,
+            dg.grid,
+            ode_solver.numerical_flux_first_order_fast,
+            dg.numerical_flux_second_order,
+            dg.numerical_flux_gradient,
+            state_auxiliary = dg.state_auxiliary,
+            state_gradient_flux = dg.state_gradient_flux,
+            states_higher_order = dg.states_higher_order,
+            direction = HorizontalDirection(),
+        )
+        fast_dg_v = DGModel(
+            fast_model,
+            dg.grid,
+            ode_solver.numerical_flux_first_order_fast,
+            dg.numerical_flux_second_order,
+            dg.numerical_flux_gradient,
+            state_auxiliary = dg.state_auxiliary,
+            state_gradient_flux = dg.state_gradient_flux,
+            states_higher_order = dg.states_higher_order,
+            direction = VerticalDirection(),
+        )
+        fast_dg = (fast_dg_h, fast_dg_v)
+        if length(ode_solver.nsubsteps) == 1
+            nsubsteps = getnsubsteps(
+                Symbol(ode_solver.mis_method),
```

Comment:
Pass the object instead of the symbol:
```suggestion
                ode_solver.mis_method,
```